common/test/common/AmpValidityTest.scala

```diff
@@ -30,16 +31,22 @@ trait AmpValidityTest extends FlatSpec with Matchers with ConfiguredTestSuite wi
       s"The AMP page at $url" should "pass an AMP validator" in getContentString(ampUrl) { content =>
 
         val commandInputWriter: OutputStream => Unit = writeToProcess(content)
+        var output = List[String]()
 
         // Generate a ProcessIO with desired input and no output (error or otherwise)
-        val io: ProcessIO = BasicIO(withIn = false, ProcessLogger((_) => ()))
+        val io: ProcessIO = BasicIO(withIn = false, ProcessLogger { line =>
+          output = line :: output // Mutate because we can't return!
+          ()
+        })
```

Comment:
How about going with `StringBuilder` or `StringBuffer` instead?

    val output = new StringBuilder

    // Generate a ProcessIO with desired input and no output (error or otherwise)
    val io: ProcessIO = BasicIO(withIn = false, ProcessLogger(line => output.append(line + "\n")))
      .withInput(commandInputWriter)

Then it's just a case of `output.toString()` for the clue.